contracts/blade/Gateway.sol

```diff
@@ -79,14 +92,37 @@ contract Gateway is ValidatorSetStorage, IGateway {
 
         verifySignature(bls.hashToPoint(DOMAIN_BRIDGE, hash), signedBridgeBatch.signature, signedBridgeBatch.bitmap);
 
+        if (block.number > signedBridgeBatch.threshold && !signedBridgeBatch.isRollback) {
```

Comment:
Using threashold should be in another PR. Remove it from this one, and open another one.